### PR TITLE
feat(death-benefits): new application

### PR DESCRIPTION
**Pull request by ylfahfa**
## What

New application - death benefits 

http://localhost:4242/umsoknir/danarbaetur/

## Why

For applicants to apply for death benefits after the loss of a spouse

/umsoknir/danarbaetur

## Gervimenn

Martha ÞÍ Ferati - 0911469
Marta ÞÍ Torp - 0501489
David ÞÍ Daniels - 1001459

## Screenshots

Personal information:
Screenshot 2024-10-30 at 14 05 08

Bank and tax details:
Screenshot 2024-10-30 at 14 06 10

Spouse info:
Screenshot 2024-10-30 at 14 06 45

Expecting child screen (if cohabitation has lasted less than a year, applicants get this question):
Screenshot 2024-10-30 at 14 07 35

Expecting child file upload screen:
Screenshot 2024-10-30 at 14 07 57

Additional documents screen:
Screenshot 2024-10-30 at 14 08 16

Additional comments:
Screenshot 2024-10-30 at 14 08 58

Application overview:
Screenshot 2024-10-30 at 14 09 43

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced a comprehensive application template for managing death benefits, including forms for prerequisites, review, and additional document uploads.
  - Added functionality to handle death benefits applications, including eligibility checks and attachment management.
  - New components for displaying children and deceased spouse information in the application review process.
  
- **Enhancements**
  - Enhanced API support for retrieving spousal information and managing attachments.
  - Improved localization support for user messages throughout the application.

- **Bug Fixes**
  - Fixed issues related to the handling of application states and transitions.

These updates significantly improve the user experience for applicants seeking death benefits, streamlining the application process and enhancing data management.

### CODE REVIEW

**Review by magnearun — APPROVED** [review]
Core files LGTM :)

**Review by Toti91 — APPROVED** [review]
Looks good

**Review by RunarVestmann — APPROVED** [review]
Stefna codeowned files look good to me

**Review by thorkellmani — APPROVED** [review]
LGTM (hugsmiðjan)

File: libs/application/templates/social-insurance-administration/death-benefits/src/lib/messages.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      id: 'db.application:social.insurance.administration.description#markdown',
+      defaultMessage:
+        'Upplýsingar um bankareikning eru sóttar á Mínar síður hjá Tryggingastofnun. \n\nTryggingastofnun sækir einungis nauðsynlegar upplýsingar til úrvinnslu umsókna og afgreiðslu mála. Ef við á þá hefur Tryggingastofnun heimild að ná í upplýsingar frá öðrum stofnunum. \n\nFrekari upplýsingar um gagnaöflunarheimild og meðferð persónuupplýsinga má finna í persónuverndarstefnu Tryggingarstofnunar [hér](https://www.tr.is/tryggingastofnun/personuvernd).',
+      description:
+        'Information about bank account will be retrieved from My Pages at the Social Insurance Administration. \n\nThe Social Insurance Administration only collects the necessary information for processing applications and determining cases. If applicable, the Social Insurance Administration is authorised to obtain information from other organisations. \n\nMore information on data collection authority and processing of personal information can be found in the privacy policy of the Insurance Administration [here](https://www.tr.is/tryggingastofnun/personuvernd).',
+    },
```
[the remark this answers is not in the document]
Comment (ylfahfa) [reply libs/application/templates/social-insurance-administration/death-benefits/src/lib/messages.ts:25]: Yes, this is the correct text as we are getting phone number and email from island.is (I will update Figma)

**Review by veronikasif — APPROVED** [review]
lgtm 🥇

**Pull request merged**